### ISSUE TITLE
How I know the weight of an already trained network is given sample?

### ISSUE THREAD

**Issue by guirighetto** [fchollet/keras#3294]
Hello I would like to take the characteristics of my last layer of a network already trained on other issues seen that one way was this:
weights = model.layers[9].get_weights()[0]

I performed a test with 420 samples in training and 210 on the test. But the weights are only 570 vectors with 500 features representing the output layer 9. I would like to know how the weights? For my idea is to get the features extracted by CNN and conduct an experiment with SVM and compare the results. Thank you in advance

**Comment by MoyanZitto** [comment]
Hum.., if I didn't misunderstand your question, what you want to express is that you have a pretrained network, and you're trying to get the weights of the last layer.

If it is what you want, maybe you should try:
```python
weights = model.layers[-1].get_weights()
```
The meaning of weights[0] depends on the type of the last layer. For conv layer, weights[0] is weights and weights[1] is bias.

(Pls correct your English carefully next time so that people around the world could understand your question better :))

**Comment by MoyanZitto** [comment]
@guirighetto 
Hum.., if I didn't misunderstand your question, what you want to express is that you have a pretrained network, and you're trying to get the weights of the last layer.

If it is what you want, maybe you should try:

weights = model.layers[-1].get_weights()

The meaning of weights[0] depends on the type of the last layer. For conv layer, weights[0] is weights and weights[1] is bias.

(Pls correct your English carefully next time so that people around the world could understand your question better :))

**Comment by guirighetto** [comment]
Thanks for answering and sorry for the English, I'm really horrible at writing.

Maybe I'm confusing really. In my problem I have 50 classes with 3 samples each. I'm using 2 samples to train and 1 sample for validation / test. My attempt is to run CNN, in which for example has 90% accuracy and extract the features of my penultimate layer for each sample, so then I can do the same experiment in SVM.

Below is my model:

    model = Sequential()
    model.add(Convolution2D(64, 5, 5, activation='relu', name='conv1_1',input_shape=(1, 48, 48), border_mode='same'))
    model.add(MaxPooling2D(pool_size=(2,2), strides=(2,2)))
    
    model.add(Convolution2D(64, 5, 5, border_mode='same'))
    model.add(Activation('relu'))
    model.add(MaxPooling2D(pool_size=(2,2), strides=(2,2)))

    model.add(Convolution2D(64, 3, 3, border_mode='same'))
    model.add(Activation('relu'))

    model.add(MaxPooling2D(pool_size=(2,2), strides=(2,2)))
    model.add(Flatten())
    model.add(Dense(500)) <- extract the feature of this layer
    model.add(Activation("relu"))
    model.add(Dropout(0.5))
    model.add(Dense(50, activation='softmax'))

You know how to do this?

I've tried this: get_feature K.function = ([model.layers [0] .input, K.learning_phase ()], [model.layers [9] .output]), but this way the model is not trained

sorry for the english again

**Comment by MoyanZitto** [comment]
As shown in your code, you want to extract the output of the "Dense(500)" layer. In this case you have two choice.
1. train your model and save the trained weights, load your model and weights in another script and use the theano/tensorflow function you showed above.
2. use Model to construct your model, make an output at the Dense(500), then train your model and get the feature from the output you just made.

**Comment by guirighetto** [comment]
Tks man
